pkg/upstream/cluster/cluster_manager.go

```diff
@@ -61,8 +61,11 @@ const globalTLSMetrics = "global"
 
 // types.ClusterManager
 type clusterManager struct {
-	clustersMap      sync.Map
-	protocolConnPool sync.Map // protocolname: { address : connpool }
+	clustersMap       sync.Map
+	clusterPoolEnable bool // TODO support modify clusterPoolEnable at runtime
+	// 0: proto: {addr: pool},
+	// 1: proto: {cluster: {addr: pool}}
+	protocolConnPool []sync.Map
```

Comment:
I suggest using an array like this https://github.com/mosn/pkg/blob/7b78cfad154db9736846cc1b2a75c3de30279d9f/internal/context/context.go#L35

Reply:
Got it , I will fix it.

Reply:
I think we can build a new strcut for `protocolConnPool`.